worldengine/draw.py

```diff
@@ -693,7 +696,7 @@ def draw_scatter_plot(world, size, target):
         for x in range(world.width):
             if world.is_land((x, y)):
                 t = world.temperature_at((x, y))
-                p = world.humidity['data'][y, x]
+                p = world.humidity_at((x, y))
```

Comment:
```
But is there any chance at all that humidity_at((x, y)) will ever not return the same value as humidity['data'][y, x]?
```
Yes: this is what happens when we change how we store data as we are doing in this PR. When we access humidity['data'] from outside world we depends on the internal structure of World, not on its interface and that means there is no encapsulation: if we change how World is organized every usage of World is affected, which is very bad from the maintanability point of view.
If instead we use proper function we can pass by having internally world.humidity to having world.layers['humidity'] without having to change any usage.

About using numpy functions we could do that through interfaces, passing lambdas or functions or exposing numpy functions (as we do with min and max which are exposed by the Layer class).

I would say we should strive for maintanability and only where there is a realy performance hit we could go for the "hack" where the hack is accessing directly internal world variables.